registry/storage/driver/middleware/redirect/middleware.go

```diff
@@ -0,0 +1,47 @@
+package middleware
+
+import (
+	"fmt"
+	"github.com/docker/distribution/context"
+	storagedriver "github.com/docker/distribution/registry/storage/driver"
+	storagemiddleware "github.com/docker/distribution/registry/storage/driver/middleware"
+	"net/url"
+	"strings"
+)
+
+type redirectStorageMiddleware struct {
+	storagedriver.StorageDriver
+	scheme string
+	host   string
+}
+
+var _ storagedriver.StorageDriver = &redirectStorageMiddleware{}
+
+func newRedirectStorageMiddleware(sd storagedriver.StorageDriver, options map[string]interface{}) (storagedriver.StorageDriver, error) {
+	o, ok := options["baseurl"]
+	if !ok {
+		return nil, fmt.Errorf("no baseurl provided")
+	}
+	b, ok := o.(string)
+	if !ok {
+		return nil, fmt.Errorf("baseurl must be a string")
+	}
+	if !strings.Contains(b, "://") {
```

Comment:
If I `url.Parse("example.com")`, the `example.com` gets stuffed into the `Path` member of `URL` instead of the `Host` member of `URL` and `Host` member is set to empty string.

I think it may be easier (and more straightforward what the scheme will be) to just straight up `Parse` the value passed in from config and check if `Scheme` is nil and `Host` is nil, thus forcing the user to specify scheme in the config.